### DISCUSSION TITLE
Problems with some Tess target pixel files

### DISCUSSION THREAD

**Discussion by cxfg** [lightkurve/lightkurve#1246] [Q&A]
Hi, as I am going through the TESS tpf data, occasionally target pixel files appear that are visibly incorrect. The exact same issue is detailed in the issues post :
https://github.com/lightkurve/lightkurve/issues/1222  by @christinahedges .

 In that post two fixes are mentioned. The first fix says said that a "check against the column and shape attributes" can be made.

Please could I get a pointer on how I would go about making this check, in order to automate the filtering of these tpf files.

Thank you

**Comment by Nschanche** [comment]
Hi there, 

As @christinahedges mentioned, this happens when you cutout a Target Pixel File (TPF) within 45 pixels of the edge of the CCD (see [this website](https://heasarc.gsfc.nasa.gov/docs/tess/data-products.html) for details on the collateral pixels). 

A TPF made using tesscut has column and row (tpf.column and tpf.row) information in the header, which gives the lower left corner pixel values. 

Given the lower left corner, you can either use the shape of your tpf to see if any of the tpf falls on collateral pixels, or just check the location of your target within the TPF by adding the 'crpix' values to the column/row:

```
target_col = tpf.column+tpf.get_header(ext=2)['crpix1']
target_row = tpf.row+tpf.get_header(ext=2)['crpix2']
```
